aws-cloudformation-stack/src/main/java/software/amazon/cloudformation/stack/DeleteHandler.java

```
@@ -0,0 +1,84 @@
+                        }
+
+                        logger.log(String.format("%s successfully deleted.", ResourceModel.TYPE_NAME));
+                        return awsResponse;
+                    })
+                    // for more information -> https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test-contract.html
+                    .stabilize((awsRequest, awsResponse, client, model, context) -> {
+                        DescribeStacksResponse describeStacksResponse;
+                        try {
+                            describeStacksResponse = client.injectCredentialsAndInvokeV2(DescribeStacksRequest.builder()
+                                .stackName(model.getStackId()).build(), client.client()::describeStacks);
+                        } catch (final AwsServiceException e) {
+                            if (e.getMessage().contains("does not exist")) {
+                                logger.log(String.format("%s %s deletion has stabilized", ResourceModel.TYPE_NAME, model.getPrimaryIdentifier()));
+                                return true;
+                            }
+                            throw new CfnGeneralServiceException(e.getMessage(), e);
+                        }
+                        if (describeStacksResponse.stacks().isEmpty()) {
+                            return true;
+                        }
+                        StackStatus status = describeStacksResponse.stacks().get(0).stackStatus();
+                        switch(status) {
+                            case DELETE_COMPLETE: {
+                                logger.log(String.format("%s [%s] deletion has stabilized", ResourceModel.TYPE_NAME, model.getPrimaryIdentifier()));
+                                return true;
+                            }
+                            case DELETE_FAILED: throw new CfnNotStabilizedException(ResourceModel.TYPE_NAME, model.getStackId());
```

did you confirm that we want to throw an error instead of retrying the delete? 

"If a DELETE_FAILED state is reached, we can either retry the delete or we can surface an error message from the API call to the customer."

---

Confirmed with Andrew, We don't retry.